packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -147,6 +162,38 @@ function trimOldestLines(
     edit.delete(document.uri, range)
 }
 
+function updateStatusBarItemsOnStreamEvent(session: LiveTailSession, event: LiveTailSessionUpdate) {
+    updateIsSampled(session, event)
+    updateEventRate(session, event)
+}
+
+function updateIsSampled(session: LiveTailSession, event: LiveTailSessionUpdate) {
+    session.isSampled =
```

Comment:
instead of setting it here for testability does it make sense to just return and then have `session.isSampled = isSampled(session, event)`?